source/application/AMPIC887DriverServer/GCS2Commands/AMGCS2AsyncGetRecordedDataValuesCommand.cpp

```diff
@@ -0,0 +1,88 @@
+#include "AMGCS2AsyncGetRecordedDataValuesCommand.h"
+
+#include "PI_GCS2_DLL.h"
+
+#include "../AMPIC887Controller.h"
+AMGCS2AsyncGetRecordedDataValuesCommand::AMGCS2AsyncGetRecordedDataValuesCommand(QVector<AMPIC887HexapodPosition> &positionData) :
+	AMGCS2AsyncCommand()
+{
+	positionData_ = &positionData;
+	rawPositionData_ = 0;
+	lastReadDataIndex_ = -2;
+	matchCounter_ = 0;
+}
+
+
+bool AMGCS2AsyncGetRecordedDataValuesCommand::runImplementation()
+{
+	positionData_->fill(AMPIC887HexapodPosition());
+	int numberOfRecordTables = 7;
+	AMCArrayHandler<int> tableIdsArray = AMCArrayHandler<int>(numberOfRecordTables);
+	AMCArrayHandler<char> header = AMCArrayHandler<char>(BUFFER_SIZE + 1);
+
+	tableIdsArray.cArray()[0] = 2;	// Position of X
+	tableIdsArray.cArray()[1] = 4;  // Position of Y
+	tableIdsArray.cArray()[2] = 6;  // Position of Z
+	tableIdsArray.cArray()[3] = 8;  // Position of U
+	tableIdsArray.cArray()[4] = 10; // Position of V
+	tableIdsArray.cArray()[5] = 12; // Position of W
+	tableIdsArray.cArray()[6] = 13; // Record time
+
+	int offsetOfFirstData = 1;
+	bool success = PI_qDRR(controller_->id(),
+						   tableIdsArray.cArray(),
+						   numberOfRecordTables,
+						   offsetOfFirstData,
+						   -1,
+						   &rawPositionData_,
+						   header.cArray(),
+						   BUFFER_SIZE + 1);
+
+	if(!success) {
+		lastError_ = controllerErrorMessage();
+	}
+
+
+	return success;
+}
+
+void AMGCS2AsyncGetRecordedDataValuesCommand::checkRunningState()
+{
+	int lastReadIndex = PI_GetAsyncBufferIndex(controller_->id());
+
+	if(lastReadIndex == -1) {
+
+		lastError_ = "Failed to retrieve any data";
+		runningState_ = Failed;
+	} else if(matchCounter_ > 10 && controller_->movementStatuses() == 0) {
```

Comment:
Maybe put a comment here that explains why we chose 10?  I feel we will be quite confused when we come back to this and are like "What the?!  10 seems really arbitrary.  I'm going to change it."